Miguel-Leite/challenge-wedev
# Add repository pattern and DTO for user entity

Summary:
This feature adds the repository pattern and DTO to the user entity.

Problem:
Currently, the user entity does not have a standard repository and DTO to manage its information.

Solution:
This feature adds a repository pattern and DTO for the user entity.

Changes:
- Add repository and DTO files to user entity
- Add seed and factory script to populate database with test data

Comments:
- No additional features have been added at this time.
- Automated tests have not been added yet.

Review Steps:
1. Check that the repository pattern and DTO are implemented correctly.
2. Verify that the seed and factory script works correctly.
